### DISCUSSION TITLE
REopt V3 API returns Status Code = 200 and empty text

### DISCUSSION THREAD

**Discussion by cpaulgilman** [NREL/REopt-API-Analysis#168] [8 - API: Application Programming Interface] [answered]
I'm trying to fix an issue we are having with the REopt API call from SAM (https://github.com/NREL/SAM/issues/1830).

The REopt API POST request that SAM constructs from the inputs in its user interface returns an empty string and Status Code = 200 instead of a UUID. I'm having trouble figuring out what's wrong with the request.

 I've attached two files:

* [reopt-test.zip](https://github.com/user-attachments/files/17603116/reopt-test.zip) contains a Jupyter notebook with a mininum code example illustrating the problem.
* [reopt_jsonpost.zip](https://github.com/user-attachments/files/17603120/reopt_jsonpost.zip) contains the complete JSON generated by SAM for the REopt API POST.

Could you help me figure out what the problem is?

I'm hoping to fix this in time for the Fall 2024 version of SAM, which we  hope to release in the next couple of weeks.

Thanks,
Paul.

**Comment by Bill-Becker** [comment]
Just updating this thread because Paul and I had back-and-forth via email. One of the key issues was a big number of 1e38 was too large for the HiGHS solver to handle by default (max of 1e15 by default). Updating the solver setting to accept a larger big number made the solve time really long, so the solution for at least that issue was to update the big number sent by SAM to REopt to 1e14. The SCIP solver was compatible with the larger big number, but that took a long time to solve as well. 

Paul - feel free to elaborate on some of the other issues that were resolved.

**Comment by cpaulgilman** [comment]
@Bill-Becker Thanks for following up. The other [change we made](https://github.com/NREL/SAM/pull/1904) to get things working was to use "stable" instead if "v3" in the URL for the REopt API call from the SAM user interface. This is not ideal for us because we want to avoid situations where an update to the stable version of REopt that breaks SAM's API call is out of sync with the SAM release schedule.

**Reply by Bill-Becker** [reply]
Please send over an example inputs.json that does not return a run_uuid with v3 - I'd like to try to debug that.

**Reply by cpaulgilman** [reply]
This JSON, [reopt_jsonpost.zip](https://github.com/user-attachments/files/17735741/reopt_jsonpost.zip), is an example generated by SAM (for the default PV-Battery / Commercial configuration) with the following results:

* **stable**: status code = 201 with a run_uuid

* **v3**: status code = 200 with no run_uuid

Here's a minimal example of Python code I'm using for this test:

```
import requests
import json

#version = "stable"
version = "v3"

post_url = "https://developer.nrel.gov/api/reopt/" + version + "/job?format=json&api_key=DEMO_KEY"

payload = open("reopt_jsonpost.json")

r = requests.post(post_url, json=json.load(payload))

print(r)
print(r.status_code)
print(r.encoding)
print(r.text)
```

Thanks,
Paul.

**Reply by Bill-Becker** [reply, marked as answer]
This is a confusing one! This works from Postman, but not from python requests. I tried to add all the same Headers that Postman was sending, for the python request. The culprit (that we're only seeing from non-Postman requests to /v3) is the urdb_response object because when I remove that it goes through with python requests to /v3.

We don't have any plans to make /v4 as of now, and we will certainly make sure you know when we even start making plans to let you know about the timeline (at least 6 months heads up!!).

**Reply by cpaulgilman** [reply]
@Bill-Becker Thanks. This works for us. We'll continue using stable and let you know if we run into any other issues.
